### PR TITLE
feat(endpoints): Expose endpoint service for plug-ins

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Endpoint service was available to extensions but not to plug-ins
Add support for plug-ins

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18874

### How to test this PR?
Add a plug-in using this extension 
FYI I have an example in this PR: https://github.com/eclipse/che-theia/pull/1024

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next

### CODE REVIEW

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-endpoint-main.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**********************************************************************
+ * Copyright (c) 2018-2021 Red Hat, Inc.
```
Comment (azatsarynnyy) [inline extensions/eclipse-che-theia-plugin-ext/src/browser/che-endpoint-main.ts:2]: why 2018?
just 2021

**Review by azatsarynnyy — APPROVED** [review]
I have not tested it. But code-wise looks good.

File: extensions/eclipse-che-theia-plugin-ext/src/browser/che-endpoint-main.ts (hunk 1 above)
Comment (benoitf) [reply extensions/eclipse-che-theia-plugin-ext/src/browser/che-endpoint-main.ts:2]: copy/paste :-) will fix

**Comment by benoitf** [comment]
@azatsarynnyy I've updated PR let me know if it's still ok (I've updated headers date and tests)

**Comment by azatsarynnyy** [comment]
> @azatsarynnyy I've updated PR let me know if it's still ok (I've updated headers date and tests)

LGTM :+1:

**Pull request merged**